src/Tests/AddressTest.php

```diff
@@ -0,0 +1,89 @@
+<?php
+declare(strict_types=1);
+
+namespace GingerPluginSdk\Tests;
+
+use GingerPluginSdk\Entities\Address;
+use GingerPluginSdk\Properties\Country;
+use GingerPluginSdk\Exceptions\OutOfEnumException;
+use GingerPluginSdk\Exceptions\OutOfPatternException;
+use GingerPluginSdk\Helpers\SingleFieldTrait;
+use PHPUnit\Framework\TestCase;
+use TypeError;
+
+class AddressTest extends TestCase
+{
+    private Address $address;
+
+    public function __construct(?string $name = null, array $data = [], $dataName = '')
+    {
+        $this->address = new Address(
+            address_type: "customer",
+            postal_code: "38714",
+            street: "Soborna",
+            city: "Poltava",
+            country: new Country(
+                "UA"
+            )
+        );
+        self::assertSame((bool)$this->address, true);
+        parent::__construct($name, $data, $dataName);
+    }
+
+    public function test_address_type_invalid_enumeration_exception()
+    {
+        self::expectException(OutOfEnumException::class);
+        $this->address->getAddressType()->set('james_bond');
```

Comment:
Why would you allow setting the address_type afterwards? It's good that you tested it but will this ever be used instead of just passing it to the constructor?

Reply:
Not sure that it would be used. In this test, i checked enum validation. Basically i want to keep tests simple and short and as i already assigned the Address object with correct data, don't want to create a new one with the same data and only 1 different field, so i just changed it.